centinel/experiments/baseline.py

```diff
@@ -0,0 +1,271 @@
+                http_netloc = url
+                http_ssl    = False
+                ssl_port = 443
+                http_path   = '/'
+
+            domain_name = http_netloc.split(':')[0]
+
+            # start tcpdump
+            td = Tcpdump()
+            tcpdump_started = False
+
+            try:
+                if self.record_pcaps:
+                    td.start()
+                    tcpdump_started = True
+                    logging.info("%s: tcpdump started..." % (url))
+                    # wait for tcpdump to initialize
+                    time.sleep(1)
+            except Exception as exp:
+                logging.warning("%s: tcpdump failed: %s" %(url, str(exp)))
+
+            # HTTP GET
+            logging.info("%s: HTTP" % (url))
+            try:
+                http_results[url] = http.get_request(http_netloc,
+                                                     http_path,
+                                                     ssl=http_ssl)
+            except Exception as exp:
+                logging.info("%s: HTTP test failed: %s" %
+                             (url, str(exp)))
+                http_results[url] = { "exception" : str(exp) }
+
+            # TLS certificate
+            # this will only work if the URL starts with https://
+            if http_ssl:
+                try:
+                    tls_result = {}
+                    logging.info("%s: TLS certificate" %
+                                 (domain_name))
+                    fingerprint, cert = tls.get_fingerprint(domain_name, ssl_port)
+                    tls_result['port'] = ssl_port
+                    tls_result['fingerprint'] = fingerprint
+                    tls_result['cert'] = cert
+
+                    tls_results[domain_name] = tls_result
+                except Exception as exp:
+                    logging.info("%s: TLS certfiticate download failed: %s" %
+                                 (domain_name, str(exp)))
+                    tls_results[domain_name] = { "exception" : str(exp) }
+
+            # DNS Lookup
+            logging.info("%s: DNS" % (domain_name))
+            try:
+                dns_results[domain_name] = dnslib.lookup_domain(domain_name)
+            except Exception as exp:
+                logging.info("%s: DNS lookup failed: %s" %
+                             (domain_name, str(exp)))
+                dns_results[domain_name] = { "exception" : str(exp) }
+
+            # Traceroute
+            for method in self.traceroute_methods:
+                try:
+                    logging.info("%s: Traceroute (%s)"
+                                 % (domain_name, method.upper()))
+                    traceroute_results[domain_name] = traceroute.traceroute(
+                        domain_name, method=method)
+                except Exception as exp:
+                    logging.info("%s: Traceroute (%s) failed: %s" %
+                                    (domain_name, method.upper(), str(exp)))
```

Comment:
you should do consistent indentation here-> move the argument tuple back 3 or so characters